pshtt/pshtt.py

```diff
@@ -526,31 +526,40 @@ def hsts_check(endpoint):
             endpoint.hsts = False
             return
 
-        endpoint.hsts = True
         endpoint.hsts_header = header
 
-        # Set max age to the string after max-age
-        # TODO: make this more resilient to pathological HSTS headers.
-
         # handle multiple HSTS headers, requests comma-separates them
-        first_pass = re.split(r',\s?', header)[0]
-        second_pass = re.sub(r'\'', '', first_pass)
+        headers = [x.strip() for x in header.split(",")]
+        # Multiple HSTS headers does not conform to RFCs 7230-3.2.2 and 6797-6.1
+        if len(headers) > 1:
+            logging.warning("Host is incorrectly returning multiple HSTS headers: {}".format(header))
+
+        # Put all of the directives in the HSTS header into a dictionary
+        directive_list = [x.strip() for x in headers[0].split(";")]
+        directives = dict()
+        for directive in directive_list:
+            components = directive.split("=")
+            directives[components[0]] = "".join(components[1:]) or True
```

Comment:
This line probably deserves a comment explaining what it does.  It's relying on the fact that the empty string evaluates to `False` and the fact that the `or` operator short circuits without forcing the whole result to be cast to a bool.

Instead of a comment, it might be clearer to break it up something like this:
```python
key = components[0]
value = "".join(components[1:])
# If the value is the empty string then there was no "=" present, and hence
# the key corresponds to a boolean flag.
if value == "":
    value = True
directives[key] = value
```

I'm fine with either way, as long as it's made clearer what is going on.

Reply:
I wasn't happy with how it looked after coming back to it so I just reworked it to make it clear what's going on.